HW04.Rmd

```diff
@@ -29,33 +29,91 @@ Knitr is part of the knitr package. Allows you to change column names and add ca
 
 
 ```{r month, warning=FALSE}
-
+number_per_month <- gun_deaths %>% 
+    select(year, month) %>% 
+    group_by(year) %>%
+    count(month) #select and count number of deaths per month
+
+number_per_month$month <- factor(number_per_month$month) 
+levels(number_per_month$month) <- c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec") #convert month to abbreviation
+    knitr::kable(x = number_per_month, format = "markdown", col.names = c("Year", "Month", "Number of Deaths"), align = "ccc", caption = "Number of Gun Deaths Per Month (2012-2014)")
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
 ```{r month_chart}
-
+number_per_month %>% 
+    ggplot(aes(x = month, y = n)) +
+    geom_col(fill = "#00CDFF") +
+    labs(title = "Number of Gun Deaths by Month (2012-2014)",
+         x = "Month", 
+         y = "Number of Deaths",
+         caption = "Source: FiveThirtyEight") +
+    theme_classic() +
+    theme(plot.title = element_text(hjust = 0.5))
+    
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
 ```{r intent}
-
+gun_deaths %>% 
+    select(intent) %>% 
+    count(intent) %>% #select and count by intent
+    arrange(desc(n)) %>% #arrange by number of deaths in descending order

```

Comment:
This is a great way to sort the bars from highest to lowest number of observations automatically! I printed a table showing the number of observations in each group, and then manually sorted the bars accordingly. This is much easier.